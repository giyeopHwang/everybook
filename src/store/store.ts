import { configureStore } from '@reduxjs/toolkit';
import { RootState } from '@/data/types';
import displayOptionsReducer from './display-options/display-options-slice';

export default configureStore<RootState>({
  reducer: {
    displayOptions: displayOptionsReducer,
  },
});
