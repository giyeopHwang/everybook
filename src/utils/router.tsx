import { createBrowserRouter } from 'react-router-dom';

import Front from '@/components/front/Front.component';
import DashboardLayout from '@/components/dashboard/dashboard-layout/DashboardLayout.compoent';

export const router = createBrowserRouter([
  {
    path: '/',
    element: <Front />,
  },
  {
    path: 'dashboard',
    element: <DashboardLayout />,
    children: [
      {
        path: '',
        element: <div>가계부</div>,
      },
      {
        path: 'report',
        element: <div>보고서</div>,
      },
      {
        path: 'budget',
        element: <div>예산 관리</div>,
      },
      {
        path: 'account',
        element: <div>분류 관리</div>,
      },
    ],
  },
]);
