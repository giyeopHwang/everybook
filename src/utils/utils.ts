import dayjs from 'dayjs';

export const displayAmount = (
  amount: number,
  showAmount: boolean = true,
  placeholder: string = '*********'
) => {
  return showAmount ? amount.toLocaleString() : placeholder;
};

export const formatDate = (date: Date = new Date()) => {
  const day = ['일', '월', '화', '수', '목', '금', '토'][dayjs(date).day()];
  return dayjs(date).format(`YYYY년 M월 D일 (${day})`);
};
