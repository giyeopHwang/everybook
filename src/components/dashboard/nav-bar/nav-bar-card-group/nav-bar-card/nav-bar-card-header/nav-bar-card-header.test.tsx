import { renderWithWrapper } from '@/test/test.utils';
import { RootState } from '@/store/store';
import { mockNet, mockPlus, mockMinus } from '@/test/test.mock';
import NavBarCardHeader from './nav-bar-card-header.component';

describe('NavBarCardHeader', () => {
  it('renders net-typed NavBarCardHeader component', () => {
    const { title, amount, type } = mockNet;
    const navBarCardHeader = renderWithWrapper(
      <NavBarCardHeader title={title} amount={amount} type={type} />
    );

    expect(navBarCardHeader).toMatchSnapshot();
  });

  it('renders plus-typed NavBarCardHeader component', () => {
    const { title, amount, type } = mockPlus;
    const navBarCardHeader = renderWithWrapper(
      <NavBarCardHeader title={title} amount={amount} type={type} />
    );

    expect(navBarCardHeader).toMatchSnapshot();
  });

  it('renders minus-typed NavBarCardHeader component', () => {
    const { title, amount, type } = mockMinus;
    const navBarCardHeader = renderWithWrapper(
      <NavBarCardHeader title={title} amount={amount} type={type} />
    );

    expect(navBarCardHeader).toMatchSnapshot();
  });

  it('renders with hidden amount', () => {
    const { title, amount, type } = mockNet;
    const initialState: Partial<RootState> = {
      displayOptions: {
        showAmount: false,
        showNavBar: true,
      },
    };

    const navBarCard = renderWithWrapper(
      <NavBarCardHeader title={title} amount={amount} type={type} />,
      { preloadedState: initialState }
    );

    expect(navBarCard).toMatchSnapshot();
  });
});
