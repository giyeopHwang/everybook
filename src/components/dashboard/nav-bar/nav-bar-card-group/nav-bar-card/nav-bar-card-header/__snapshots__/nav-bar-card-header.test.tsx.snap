// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavBarCardHeader > renders minus-typed NavBarCardHeader component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-pe6vce MuiBox-root css-1y16nxq"
      >
        <div
          class="css-44nm6q MuiBox-root css-0"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1u2zctn-MuiSvgIcon-root"
            data-testid="RemoveIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13H5v-2h14v2z"
            />
          </svg>
          <p
            class="MuiTypography-root MuiTypography-body2 css-1d3njat-MuiTypography-root"
          >
            MINUS
          </p>
        </div>
        <p
          class="MuiTypography-root MuiTypography-body2 css-1d3njat-MuiTypography-root"
        >
          200
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-pe6vce MuiBox-root css-1y16nxq"
    >
      <div
        class="css-44nm6q MuiBox-root css-0"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1u2zctn-MuiSvgIcon-root"
          data-testid="RemoveIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13H5v-2h14v2z"
          />
        </svg>
        <p
          class="MuiTypography-root MuiTypography-body2 css-1d3njat-MuiTypography-root"
        >
          MINUS
        </p>
      </div>
      <p
        class="MuiTypography-root MuiTypography-body2 css-1d3njat-MuiTypography-root"
      >
        200
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`NavBarCardHeader > renders net-typed NavBarCardHeader component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-pe6vce MuiBox-root css-16qv2i2"
      >
        <div
          class="css-44nm6q MuiBox-root css-0"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1u2zctn-MuiSvgIcon-root"
            data-testid="TollIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15 4c-4.42 0-8 3.58-8 8s3.58 8 8 8 8-3.58 8-8-3.58-8-8-8zm0 14c-3.31 0-6-2.69-6-6s2.69-6 6-6 6 2.69 6 6-2.69 6-6 6zM3 12c0-2.61 1.67-4.83 4-5.65V4.26C3.55 5.15 1 8.27 1 12s2.55 6.85 6 7.74v-2.09c-2.33-.82-4-3.04-4-5.65z"
            />
          </svg>
          <p
            class="MuiTypography-root MuiTypography-body2 css-1d3njat-MuiTypography-root"
          >
            NET
          </p>
        </div>
        <p
          class="MuiTypography-root MuiTypography-body2 css-1d3njat-MuiTypography-root"
        >
          100
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-pe6vce MuiBox-root css-16qv2i2"
    >
      <div
        class="css-44nm6q MuiBox-root css-0"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1u2zctn-MuiSvgIcon-root"
          data-testid="TollIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15 4c-4.42 0-8 3.58-8 8s3.58 8 8 8 8-3.58 8-8-3.58-8-8-8zm0 14c-3.31 0-6-2.69-6-6s2.69-6 6-6 6 2.69 6 6-2.69 6-6 6zM3 12c0-2.61 1.67-4.83 4-5.65V4.26C3.55 5.15 1 8.27 1 12s2.55 6.85 6 7.74v-2.09c-2.33-.82-4-3.04-4-5.65z"
          />
        </svg>
        <p
          class="MuiTypography-root MuiTypography-body2 css-1d3njat-MuiTypography-root"
        >
          NET
        </p>
      </div>
      <p
        class="MuiTypography-root MuiTypography-body2 css-1d3njat-MuiTypography-root"
      >
        100
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`NavBarCardHeader > renders plus-typed NavBarCardHeader component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-pe6vce MuiBox-root css-c1a362"
      >
        <div
          class="css-44nm6q MuiBox-root css-0"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1u2zctn-MuiSvgIcon-root"
            data-testid="AddIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
          <p
            class="MuiTypography-root MuiTypography-body2 css-1d3njat-MuiTypography-root"
          >
            PLUS
          </p>
        </div>
        <p
          class="MuiTypography-root MuiTypography-body2 css-1d3njat-MuiTypography-root"
        >
          300
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-pe6vce MuiBox-root css-c1a362"
    >
      <div
        class="css-44nm6q MuiBox-root css-0"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1u2zctn-MuiSvgIcon-root"
          data-testid="AddIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
        </svg>
        <p
          class="MuiTypography-root MuiTypography-body2 css-1d3njat-MuiTypography-root"
        >
          PLUS
        </p>
      </div>
      <p
        class="MuiTypography-root MuiTypography-body2 css-1d3njat-MuiTypography-root"
      >
        300
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`NavBarCardHeader > renders with hidden amount 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-pe6vce MuiBox-root css-16qv2i2"
      >
        <div
          class="css-44nm6q MuiBox-root css-0"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1u2zctn-MuiSvgIcon-root"
            data-testid="TollIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15 4c-4.42 0-8 3.58-8 8s3.58 8 8 8 8-3.58 8-8-3.58-8-8-8zm0 14c-3.31 0-6-2.69-6-6s2.69-6 6-6 6 2.69 6 6-2.69 6-6 6zM3 12c0-2.61 1.67-4.83 4-5.65V4.26C3.55 5.15 1 8.27 1 12s2.55 6.85 6 7.74v-2.09c-2.33-.82-4-3.04-4-5.65z"
            />
          </svg>
          <p
            class="MuiTypography-root MuiTypography-body2 css-1d3njat-MuiTypography-root"
          >
            NET
          </p>
        </div>
        <p
          class="MuiTypography-root MuiTypography-body2 css-1d3njat-MuiTypography-root"
        >
          *********
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-pe6vce MuiBox-root css-16qv2i2"
    >
      <div
        class="css-44nm6q MuiBox-root css-0"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1u2zctn-MuiSvgIcon-root"
          data-testid="TollIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15 4c-4.42 0-8 3.58-8 8s3.58 8 8 8 8-3.58 8-8-3.58-8-8-8zm0 14c-3.31 0-6-2.69-6-6s2.69-6 6-6 6 2.69 6 6-2.69 6-6 6zM3 12c0-2.61 1.67-4.83 4-5.65V4.26C3.55 5.15 1 8.27 1 12s2.55 6.85 6 7.74v-2.09c-2.33-.82-4-3.04-4-5.65z"
          />
        </svg>
        <p
          class="MuiTypography-root MuiTypography-body2 css-1d3njat-MuiTypography-root"
        >
          NET
        </p>
      </div>
      <p
        class="MuiTypography-root MuiTypography-body2 css-1d3njat-MuiTypography-root"
      >
        *********
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
